sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -343,5 +345,17 @@ class FileContextBasedCheckpointFileManager(path: Path, hadoopConf: Configuratio
     case _: LocalFs | _: RawLocalFs => true // LocalFs = RawLocalFs + ChecksumFs
     case _ => false
   }
+
+  private def mayRemoveCrcFile(path: Path): Unit = {
+    val checksumFile = new Path(path.getParent, s".${path.getName}.crc")
```

Comment:
I just found one bug in this Path API: https://issues.apache.org/jira/browse/HDFS-14762
How about wrap the whole block like this
```
try {
  val checksumFile = new Path(path.getParent, s".${path.getName}.crc")
  if (exists(checksumFile)) {
        // checksum file exists, deleting it
        delete(checksumFile)
      }
} catch {
  case NonFatal(_) => ...
}
```

Reply:
Nice finding! HDFS-14762 looks odd but we have to live with it. The suggestion looks great.